src/main/java/frc/robot/subsystems/swerve/module/maple/MapleModule.java

```diff
@@ -0,0 +1,123 @@
+	private final PIDController steerPIDControllerRadians;
+	private final PIDController drivePIDControllerRadiansPerSecond;
+	private final SimpleMotorFeedforward driveFeedForwardRadiansPerSecond;
+
+	public MapleModule(ModuleConstants constants, MapleModuleConstants mapleModuleConstants, SwerveModuleSimulation moduleSimulation) {
+		super(constants);
+		this.moduleSimulation = moduleSimulation;
+		this.moduleIOInputs = new ModuleIOInputsAutoLogged();
+		this.steerPIDControllerRadians = mapleModuleConstants.steerPIDControllerRadians();
+		this.drivePIDControllerRadiansPerSecond = mapleModuleConstants.drivePIDControllerRadiansPerSecond();
+		this.driveFeedForwardRadiansPerSecond = mapleModuleConstants.driveFeedForwardRadiansPerSecond();
+	}
+
+	@Override
+	public SysIdCalibrator.SysIdConfigInfo getSteerSysIdConfigInfo() {
+		return new SysIdCalibrator.SysIdConfigInfo(new SysIdRoutine.Config(), false);
+	}
+
+	@Override
+	public SysIdCalibrator.SysIdConfigInfo getDriveSysIdConfigInfo() {
+		return new SysIdCalibrator.SysIdConfigInfo(new SysIdRoutine.Config(), false);
+	}
+
+	@Override
+	public void updateInputs() {
+		moduleIOInputs.drivePositionRad = moduleSimulation.getDriveWheelFinalPositionRad();
+		moduleIOInputs.driveVelocityRadPerSec = moduleSimulation.getDriveWheelFinalSpeedRadPerSec();
+		moduleIOInputs.driveAppliedVolts = moduleSimulation.getDriveMotorAppliedVolts();
+		moduleIOInputs.driveCurrentAmps = new double[] {Math.abs(moduleSimulation.getDriveMotorSupplyCurrentAmps())};
+
+		moduleIOInputs.turnAbsolutePosition = moduleSimulation.getSteerAbsoluteFacing();
+		moduleIOInputs.turnPosition = Rotation2d.fromRadians(moduleSimulation.getSteerRelativeEncoderPositionRad());
+		moduleIOInputs.turnVelocityRadPerSec = moduleSimulation.getSteerRelativeEncoderSpeedRadPerSec();
+		moduleIOInputs.turnAppliedVolts = moduleSimulation.getSteerMotorAppliedVolts();
+		moduleIOInputs.turnCurrentAmps = new double[] {Math.abs(moduleSimulation.getSteerMotorSupplyCurrentAmps())};
+
+		Logger.processInputs(constants.logPath() + "/moduleIO", moduleIOInputs);
+	}
+
+	@Override
+	public void setBrake(boolean brake) {}
+
+	@Override
+	public void resetByEncoder() {}
+
+	@Override
+	public void setDriveVoltage(double voltage) {
+		setClosedLoop(false);
+		moduleSimulation.requestDriveVoltageOut(voltage);
+	}
+
+	@Override
+	public void setSteerVoltage(double voltage) {
+		moduleSimulation.requestSteerVoltageOut(voltage);
+	}
+
+	@Override
+	public void setTargetSteerPosition(Rotation2d targetSteerPosition) {
+		setSteerVoltage(steerPIDControllerRadians.calculate(getSteerLatestPosition().getRadians(), targetSteerPosition.getRadians()));
+	}
+
+	@Override
+	public void setTargetClosedLoopVelocity(double targetVelocityMetersPerSecond) {
+		Rotation2d targetVelocityRadiansPerSecond = fromDriveMeters(targetVelocityMetersPerSecond);
+		double voltage = driveFeedForwardRadiansPerSecond.calculate(targetVelocityRadiansPerSecond.getRadians())
```

Comment:
looks like it's deprecated... I get the following warning:
> ```Note: Some input files use or override a deprecated API.```

Reply:
you're right its deprecated but the fix is not for this pr